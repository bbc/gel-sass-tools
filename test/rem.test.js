describe('rem()', () => {
  describe('function', () => {
    it('should return pixel values converted to rem', async () => {
      const css = await global.compileCSS('test/data/rem/rem-only.scss');

      expect(css).toMatchSnapshot();
    });
  });

  describe('mixin', () => {
    it('should return pixel values, and rem values', async () => {
      const css = await global.compileCSS('test/data/rem/mixin-px-and-rem.scss');

      expect(css).toMatchSnapshot();
    });

    it('should return rem only values, when pixel fallback is disabled', async () => {
      const css = await global.compileCSS('test/data/rem/mixin-rem-only.scss');

      expect(css).toMatchSnapshot();
    });
  });
});
