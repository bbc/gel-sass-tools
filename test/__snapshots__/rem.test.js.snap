// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rem() function should return pixel values converted to rem 1`] = `
".demo-1 {
  font-size: 1rem;
}

.demo-2 {
  font-size: 1.5rem;
}

.demo-3 {
  font-size: 2rem;
}"
`;

exports[`rem() mixin should return pixel values, and rem values 1`] = `
".demo-1 {
  font-size: 16px;
  font-size: 1rem;
}

.demo-2 {
  font-size: 24px;
  font-size: 1.5rem;
}

.demo-3 {
  font-size: 32px;
  font-size: 2rem;
}"
`;

exports[`rem() mixin should return rem only values, when pixel fallback is disabled 1`] = `
".demo-1 {
  font-size: 1rem;
}

.demo-2 {
  font-size: 1.5rem;
}

.demo-3 {
  font-size: 2rem;
}"
`;
